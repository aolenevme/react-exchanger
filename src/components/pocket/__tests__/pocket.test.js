import React from "react";
import renderer from "react-test-renderer";
import constant from "lodash/constant.js";

import Pocket from "../pocket.js";

describe("<Pocket />", () => {
    it("is not rendered", () => {
        expect(renderer
            .create(<Pocket/>)
            .toJSON()).toMatchSnapshot();
        expect(renderer
            .create(<Pocket currency=""/>)
            .toJSON()).toMatchSnapshot();
        expect(renderer
            .create(<Pocket currency="USD" input={constant(false)}/>)
            .toJSON()).toMatchSnapshot();
    });

    it("is partially rendered", () => {
        expect(renderer
            .create(<Pocket currency="USD" input={constant(<input />)}/>)
            .toJSON()).toMatchSnapshot();
    });

    it("is fully rendered", () => {
        expect(renderer
            .create(<Pocket currency="USD" input={constant(<input />)} balance="You have $51.12" rate="1$ = 0.69Â£"/>)
            .toJSON()).toMatchSnapshot();
    });
});
