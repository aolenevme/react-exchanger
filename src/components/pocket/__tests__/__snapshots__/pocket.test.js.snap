// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pocket /> is fully rendered 1`] = `
.c4 {
  font-size: 0.8rem;
  color: #7199e6;
}

.c2 {
  font-size: 3rem;
  line-height: 3rem;
  color: #f3f6ff;
}

.c0 {
  padding: 2rem;
  width: 100%;
  box-sizing: border-box;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 1rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 0.3rem;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <span
      className="c2"
    >
      USD
    </span>
    <input />
  </div>
  <div
    className="c3"
  >
    <span
      className="c4"
    >
      You have $51.12
    </span>
    <span
      className="c4"
    >
      1$ = 0.69Â£
    </span>
  </div>
</div>
`;

exports[`<Pocket /> is not rendered 1`] = `null`;

exports[`<Pocket /> is not rendered 2`] = `null`;

exports[`<Pocket /> is not rendered 3`] = `null`;

exports[`<Pocket /> is partially rendered 1`] = `
.c2 {
  font-size: 3rem;
  line-height: 3rem;
  color: #f3f6ff;
}

.c0 {
  padding: 2rem;
  width: 100%;
  box-sizing: border-box;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 1rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 0.3rem;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <span
      className="c2"
    >
      USD
    </span>
    <input />
  </div>
  <div
    className="c3"
  />
</div>
`;
